01-data-structures/03-linked-lists/linked_list.rb

```diff
@@ -4,27 +4,120 @@ class LinkedList
   attr_accessor :head
   attr_accessor :tail
 
+  def initialize
+    @head = nil
+    @tail = nil
+  end
+
+  def find_index(number)
+    current = @head
+    count = 1
+    until count == number
+      current = current.next
+      count += 1
+    end
+    current
+  end
+
+  def delete_index(number)
+    current = @head
+    count = 1
+    until count == number
+      current = current.next
+      count += 1
+    end
+    self.delete(current)
+  end
+
   # This method creates a new `Node` using `data`, and inserts it at the end of the list.
   def add_to_tail(node)
+    if @tail.nil?
+      @head = node
+    else
+      @tail.next = node
+    end
+    @tail = node
   end
 
   # This method removes the last node in the lists and must keep the rest of the list intact.
   def remove_tail
+    return nil if @head == nil
+
+    node = @head
+    popped = @tail
+    if @head == @tail
+      @head = nil
+      @tail = nil
+    else
+      until node.next == @tail
```

Comment:
default case should go first

Reply:
you mean
```
else
  until @tail == node.next
```
?

Reply:
most of the time the head won't be the tail, yes